internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -1058,3 +1072,123 @@ func (a *Adapter) findSnapshotWithOpenedPR(snapshot *applicationapiv1alpha1.Snap
+					a.logger.Error(err, "Failed to mark snapshot as canceled", "snapshot.Name", snapshot.Name)
+					return err
+				}
+				// now get all integration pipelineruns for a snapshot
+				integrationTestPipelineruns, err := getAllIntegrationPipelineRunsForSnapshot(a.context, a.client, &sortedSnapshots[i])
+				if err != nil {
+					a.logger.Error(err, "Failed to get all integration pipelineruns for snapshot", "snapshot.Name", snapshot.Name)
+					return err
+				}
+				for _, plr := range integrationTestPipelineruns {
+					plr := plr
+					pipelinerunStatus, _, err := h.GetRunningIntegrationPipelineRunStatus(a.context, a.client, &plr)
+					if err != nil {
+						a.logger.Error(err, "Failed to get integration test pipelinerun status", "snapshot.Name", plr.Name)
+						return err
+					}
+					if pipelinerunStatus == intgteststat.IntegrationTestStatusInProgress {
+						//remove finalizer and cancel pipelinerun
+						err = h.RemoveFinalizerFromPipelineRun(a.context, a.client, a.logger, &plr, h.IntegrationPipelineRunFinalizer)
+						if err != nil {
+							a.logger.Error(err, "Failed to remove finalizer from integration test pipelinerun")
+							return err
+						}
+						// set "CancelledRunFinally" to PLR status, should gracefully cancel pipelinerun, this is so raw I hate this
+						patch := client.MergeFrom(plr.DeepCopy())
+						plr.Spec.Status = tektonv1.PipelineRunSpecStatusCancelledRunFinally
+						err := a.client.Status().Patch(a.context, &plr, patch)
+						if err != nil {
+							return err
+						}
+
+					}
+				}
+			}
+		}
+		if gitops.IsComponentSnapshot(&sortedSnapshots[i]) {
+
+			if snapshot.GetAnnotations()[gitops.PipelineAsCodeRepoURLAnnotation] == sortedSnapshots[i].GetAnnotations()[gitops.PipelineAsCodeRepoURLAnnotation] && snapshot.GetAnnotations()[gitops.PipelineAsCodePullRequestAnnotation] == sortedSnapshots[i].GetAnnotations()[gitops.PipelineAsCodePullRequestAnnotation] {
+				err = gitops.MarkSnapshotAsCanceled(a.context, a.client, &sortedSnapshots[i], "Snapshot canceled/superseded")
+				if err != nil {
+					a.logger.Error(err, "Failed to mark snapshot as canceled", "snapshot.Name", snapshot.Name)
+					return err
+				}
+				// get all integration pipelineruns for a snapshot
+				integrationTestPipelineruns, err := getAllIntegrationPipelineRunsForSnapshot(a.context, a.client, &sortedSnapshots[i])
+				if err != nil {
+					a.logger.Error(err, "Failed to get all integration pipelineruns for snapshot", "snapshot.Name", snapshot.Name)
+					return err
+				}
+				for _, plr := range integrationTestPipelineruns {
+					plr := plr
+					pipelinerunStatus, _, err := h.GetRunningIntegrationPipelineRunStatus(a.context, a.client, &plr)
+					if err != nil {
+						a.logger.Error(err, "Failed to get integration test pipelinerun status", "snapshot.Name", plr.Name)
+						return err
+					}
+					if pipelinerunStatus == intgteststat.IntegrationTestStatusInProgress {
+						//remove finalizer and cancel pipelinerun
```

Comment:
Why delete the finalizer for the in progress PLR only? I think, once new PLR is triggered, then all older PLRs should not be kept. HDYT?

Reply:
This should be the older pipelinerun, which is still running and we dont need it anymore.

Reply:
I think the completed older integration PLRs are not needed any more. Ignore this comment. Thanks.

Reply:
This makes me think we may ignore this type of cancelled snapshot in statusreport controller or snapshot controller as well in epic [STONEINTG-1049](https://issues.redhat.com/browse/STONEINTG-1049), HDYT? cc @dirgim

Reply:
That's an interesting idea, it might be beneficial to skip processing cancelled Snapshots in the statusreport controller at least.